command/login.go

```diff
@@ -372,3 +368,14 @@ func (c *LoginCommand) extractToken(client *api.Client, secret *api.Secret, unwr
 		return nil, false, fmt.Errorf("no auth or wrapping info in response")
 	}
 }
+
+// Warn if the VAULT_TOKEN environment variable is set, as that will take
+// precedence. We output as a warning, so piping should still work since it
+// will be on a different stream.
+func (c *LoginCommand) checkForAndWarnAboutLoginToken() {
```

Comment:
nice!

Reply:
Agreed, this is great addition! I could imagine having a token set on the environment to in conjunction with issuing a login attempt to be confusing.